src/html5/parser/document.rs

```diff
@@ -692,6 +727,103 @@ impl DocumentHandle {
 
         Ok(())
     }
+
+    fn check_query_predicate(
```

Comment:
Could probably think of a better name for this... maybe `fn matches_query_condition`